src/latch_cli/nextflow/workflow.py

```diff
@@ -226,29 +218,101 @@ def _get_flags_for_dataclass(name: str, val: Any) -> List[str]:
     return flags
 
 
-def get_flag(name: str, val: Any) -> List[str]:
+def get_flag(name: str, val: object) -> list[str]:
     flag = f"--{name}"
 
     if val is None:
         return []
-    elif isinstance(val, bool):
+    if isinstance(val, bool):
         return [flag] if val else []
-    elif isinstance(val, LatchFile) or isinstance(val, LatchDir):
+    if isinstance(val, (LatchFile, LatchDir)):
         if val.remote_path is not None:
             return [flag, val.remote_path]
 
         return [flag, str(val.path)]
-    elif is_dataclass(val):
+    if is_dataclass(val):
         return _get_flags_for_dataclass(name, val)
-    elif isinstance(val, Enum):
-        return [flag, getattr(val, "value")]
-    else:
-        return [flag, str(val)]
+    if isinstance(val, Enum):
+        return [flag, val.value]
+
+    return [flag, str(val)]
+
+
+def is_samplesheet_parameter(f: Field[object]) -> bool:
+    origin = get_origin(f.type)
+    if origin is not Annotated:
+        return False
+
+    args = get_args(f.type)
+    assert len(args) > 1
+
+    ann = args[1]
+    if not isinstance(ann, FlyteAnnotation):
```

Comment:
the proper way to do this is to search through all annotations to find the type you are looking for, it does not have to be first